File: MachineCoding/ParkingLot/ParkingLot/src/main/java/org/example/ParkingSpot.java
package org.example;

public class ParkingSpot {
    String parkingSpotID = "";
    VehicleType spotType;
    int floorNumber = -1;
    Status status = Status.Free;

    public String getParkingSpotID() {
        return parkingSpotID;
    }

    public void setParkingSpotID(String parkingSpotID) {
        this.parkingSpotID = parkingSpotID;
    }

    public VehicleType getSpotType() {
        return spotType;
    }

    public void setSpotType(VehicleType spotType) {
        this.spotType = spotType;
    }

    public int getFloorNumber() {
        return floorNumber;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public void setFloorNumber(int floorNumber) {
        this.floorNumber = floorNumber;
    }
}

File: devpractice/machinecoding/ParkingLot/ParkingLot/ParkingSpot.cs
namespace ParkingLot
{
    public class ParkingSpot
    {
        public string id { get; set; }
        public int floorNum { get; set; }
        public VehicleType vehicleType { get; set; }
        public Status status { get; set; }
        public ParkingSpot(string id, int floorNum, VehicleType vehicleType, Status status)
        {
            this.id = id;
            this.floorNum = floorNum;
            this.vehicleType = vehicleType;
            this.status = status;
        }
    }
}
